Assets/Scripts/GhostGlimpse/GhostGlimpseManager.cs

```diff
@@ -0,0 +1,117 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GhostGlimpseManager : MonoBehaviour
+{
+    [SerializeField]
+    private float glimpseFrequencyBase;
+    [SerializeField]
+    private float glimpseFrequency;
+
+    private float closestTriggerRefreshBase = 0.5f;
+    [SerializeField]
+    private float closestTriggerRefresh;
+
+    [SerializeField]
+    private float glimpseFrequencyVariation;
+
+    [SerializeField]
+    private Transform playerTransform;
+
+    private Transform[] glimpseTriggers;
+
+    private Transform currentlyActiveTrigger = null;
+    // Start is called before the first frame update
+    void Start()
+    {
+        glimpseFrequency = glimpseFrequencyBase;
+        closestTriggerRefresh = 0f;
+
+        glimpseTriggers = new Transform[transform.childCount];
+        //Initialzing list of children
+        for (int i = 0; i < transform.childCount; i++)
+        {
+            glimpseTriggers[i] = transform.GetChild(i);
+            glimpseTriggers[i].GetComponent<GhostGlimpseTrigger>().used = true;
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        //Activating a glimpse trigger
+        glimpseFrequency -= Time.deltaTime;
+        if(glimpseFrequency < 0)
+        {
+            //updates active trigger after a certain amount of time
+            closestTriggerRefresh -= Time.deltaTime;
+            if(closestTriggerRefresh < 0)
+            {
+                ActivateClosestTrigger();
+                closestTriggerRefresh = closestTriggerRefreshBase;
+            }
+
+            //If trigger was hit reset the glimpse timer
+            if(currentlyActiveTrigger.GetComponent<GhostGlimpseTrigger>().used == true)
+            {
+                //Resetting glimpse timer
+                glimpseFrequency = glimpseFrequencyBase + (glimpseFrequencyVariation * Random.value);
+                closestTriggerRefresh = 0;
+            }
+
+        }
+    }
+
+    private Transform ActivateClosestTrigger()
+    {
+        //Finds the closest trigger
+        float smallestDist = float.MaxValue;
+        float currentdist;
+        Transform currentClosestTrigger = null;
+
+        for (int i = 0; i < glimpseTriggers.Length; i++)
+        {
+            Transform currentTrigger = glimpseTriggers[i];
+            currentdist = (playerTransform.position - currentTrigger.position).magnitude;
```

Comment:
Vector3.magnitude is not the same as the distance, it is the **size** of the vector. The square root of the magnitude however is equal to the distance. You can get this value via the property Vector3.sqrMagnitude.

Reply:
I'm not sure if I got something wrong but I printed a Vector3.magnitude and used Vector3.Distance() and I got the same value so I'm going to just use Vector3.Distance(). Also Vector3.sqrMagnitude gives the square of the magnitude, not the square root but I understand what you're trying to say.